src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -237,20 +236,26 @@ public bool TryAddModelError(string key, Exception exception, ModelMetadata meta
             if (exception is FormatException || exception is OverflowException)
             {
                 // Convert FormatExceptions and OverflowExceptions to Invalid value messages.
-                ModelStateEntry entry;
-                TryGetValue(key, out entry);
+                TryGetValue(key, out var entry);
 
-                var name = metadata.GetDisplayName();
+                var messageProvider = metadata.ModelBindingMessageProvider;
+                var name = metadata.DisplayName ?? metadata.PropertyName;
                 string errorMessage;
-                if (entry == null)
+                if (entry == null && name == null)
```

Comment:
Think I'll go w/
``` c#
// Not using metadata.GetDisplayName() or a single resource to avoid strange messages like
// "The value '' is not valid." (when no value was provided, not even an empty string) and
// "The supplied value is invalid for Int32." (when error is for an element or parameter).
```